File: Web/LogsAndAuth/logs_and_auth/lib/Models/UserModel.dart
class UserModel {
  int? Id;
  String? Login;
  String? Name;
  String? Email;
  String? Password;
  String? Phone;

  UserModel({ this.Id,this.Login,this.Name,this.Email,this.Password,this.Phone});

  UserModel.fromJson(Map<String, dynamic> json) {
    Id = json['id'];
    Login = json['login'];
    Name = json['name'];
    Email = json['mail'];
    Password = json['password'];
    Phone = json['phone_number'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.Id;
    data['login'] = this.Login;
    data['name'] = this.Name;
    data['mail'] = this.Email;
    data['password'] = this.Password;
    data['phone_number'] = this.Phone;
    return data;
  }
}

File: Web/Auth/Api/ApiForAuth/ApiForAuth/Models/UserModel.cs
namespace ApiForAuth.Models
{
    public class UserResponse
    {
        public int Id { get; set; }
        public int Code { get; set; }
        public string Message { get; set; }
        public UserData Data { get; set; }
        

        public static UserResponse Success(string message, UserData data, string token = null)
        {
            return new UserResponse
            {
                Id = 1,
                Code = 0,
                Message = message,
                Data = data,
                
            };
        }

        public static UserResponse Error(string message)
        {
            return new UserResponse
            {
                Id = 1,
                Code = 1,
                Message = "Error: " + message,
                Data = null
               
            };
        }
    }

    public class UserData
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Token { get; set; }
    }

    public class UserAuth
    {
        public string Login { get; set; }
        public string Password { get; set; }
    }
    public class UserModel
    {
        public string Login { get; set; }
        public string Password { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public string Mail { get; set; }
        public string NumberPhone { get; set; }
    }

}




